test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -102,9 +110,427 @@ spec = do
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected [a], encountered String"
+            ]
+
+    it "WALLETS_CREATE_05 - Num as mnemonic_sentence -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": "123",
+                "mnemonic_sentence": 15,
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected [a], encountered Number"
+            ]
+
+    it "WALLETS_CREATE_05 - mnemonic_sentence param missing -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": "A name",
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "key \"mnemonic_sentence\" not present"
+            ]
+
+    describe "WALLETS_CREATE_06 - Mnemonics second factor" $ do
+        let matrix =
+                [ ( "[] as mnemonic_second_factor -> fail", []
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid number of words: 9 or 12\
+                            \ words are expected."
+                     ]
+                   )
+                 , ( "specMnemonicSecondFactor -> fail", specMnemonicSecondFactor
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid entropy checksum: please\
+                         \ double-check the last word of your mnemonic sentence."
+                     ]
+                   )
+                 , ( "invalid mnemonics -> fail", invalidMnemonics12
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid number of words: 9 or 12\
+                         \ words are expected."
+                     ] -- why not Invalid entropy checksum ?
+                   )
+                 , ( "Japanese mnemonics -> fail", japaneseMnemonics12
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid number of words: 9 or 12\
+                          \ words are expected."
+                     ]
+                   )
+                 , ( "Chinese mnemonics -> fail", chineseMnemonics9
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Found invalid (non-English) word:"
+                     ] -- why only for Chinese?
```

Comment:
again `Found invalid (non-English) word:`. This time for Chinese mnemonics...
for Japanese and French it throws `Invalid number of words:`